OpenRA.Mods.Common/Orders/BeaconOrderGenerator.cs

```diff
@@ -30,7 +32,12 @@ protected override IEnumerable<Order> OrderInner(World world, CPos cell, int2 wo
 		protected override IEnumerable<IRenderable> RenderAnnotations(WorldRenderer wr, World world) { yield break; }
 		protected override string GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi)
 		{
-			return "ability"; // TODO: [CursorReference]
+			var beacon = world.LocalPlayer.PlayerActor.Info.TraitInfoOrDefault<PlaceBeaconInfo>();
```

Comment:
`TraitInfo<PlaceBeaconInfo>()` will suffice here. It'll throw if the trait info is missing. We can reasonably expect to only be in the `BeaconOrderGenerator` when placing a beacon, so we don't need to put as much effort into the error message as the Lua script handling does.